### DISCUSSION TITLE
Proposed V2.0 Changes

### DISCUSSION THREAD

**Discussion by m-markovic** [TEC-Toolkit/ECFO#30] [General]
General tracker for changes please reply to individual comments about specific items

### Items for Discussion

- [ ] Item 1: Review ecfo:hasEmissionSource and ecfo:hasEmissionTarget properties
- [ ] Item 2: Handle GWP Conversion Factors
- [ ] Item 3: Handle Dimensionless Conversion Factors (e.g., fractions)
- [ ] Item 4: Expand modelling of activities
- [ ] Item 5: Handle links between conversion factors (e.g., conversion factors for individual portions of other gases linked to an aggregate Co2e conversion factor) 
- [ ] Item 6: Review the ecfo:hasTag property

**Comment by m-markovic** [comment]
### Item 1: Review `ecfo:hasEmissionSource` and `ecfo:hasEmissionTarget` properties

To something like this:

```mermaid
graph LR
    EmissionConversionFactor -- hasSource --> Element
    EmissionConversionFactor -- hasTarget --> Element
    Element -- hasUnit --> Unit
    Element -- hasMatter --> Matter
```

or like this:

```mermaid
graph LR
    EmissionConversionFactor -- hasSource --> Source
    EmissionConversionFactor -- hasTarget --> Target
    Source -- rdfs:subClassOf --> Element
    Target-- rdfs:subClassOf --> Element
    Element -- hasUnit --> Unit
    Element -- hasMatter --> Matter
```

**Reply by dgarijo** [reply]
This is just a simpler design, but I think it's not a super priority

**Reply by m-markovic** [reply]
Perhaps we should remove the reference to "emission" in the name of the property to accommodate the conversion factors that convert for example one type of emissions into another type (e.g., N2O -> N2O-N, CH4-> CO2e, etc.) and not just activity data into emissions.

**Reply by stefanogermano** [reply]
On the "operational" side, I suggest to include a brief 'description' to each new comment about an item for discussion (in addition to the 'title') in order to ensure everyone is on the same page and minimize potential misunderstandings.
I did it for this one, but please feel free to modify it if you meant something different.

**Reply by stefanogermano** [reply]
In this case, I think the suggested change could not only help to have a better structure, but also to facilitate the development of a more advanced representation for some of these aspects, potentially helping with some issues we currently have (e.g., IPCC units).

**Reply by dgarijo** [reply]
I would not remove "emission" as of now. It's a major rename of the project. 
The ontology URIs, the w3ids... everything is impacted by it.

**Reply by m-markovic** [reply]
Just to clarify I was suggesting only to rename the properties e.g., ecfo:hasEmissionSource to ecfo:hasSource or similar

**Comment by m-markovic** [comment]
Item 2: Handle GWP Conversion Factors

**Reply by dgarijo** [reply]
We are working on this. Will have a solution soon, importing 6-10 GWP factors and associating them to the corresponding conversion factors. Also, will fix the "N2O of CO2E" by adding that as a new quantity and linking the GWP

**Comment by m-markovic** [comment]
Handle Dimensionless Conversion Factors (e.g., fractions)

**Comment by m-markovic** [comment]
Item 4: Expand modelling of activities

**Comment by m-markovic** [comment]
Item 5: Handle links between conversion factors (e.g., conversion factors for individual portions of other gases linked to an aggregate Co2e conversion factor)

**Reply by dgarijo** [reply]
We are working on this. Will push soon

**Comment by m-markovic** [comment]
Item 6 : Review the ecfo:hasTag property

I propose this property to change to ecfo:hasCategory with range skos:Concept (or ecfo:Category which is a subclass of skos:Concept as shown in the Formal / Semi-Formal Hybrids -- Part OWL, Part SKOS pattern in https://www.w3.org/2006/07/SWD/SKOS/skos-and-owl/master.html#Going1). This then enables to link ECF to taxonomies of different categories (e.g., the IPCC 2006 https://www.ipcc-nggip.iges.or.jp/EFDB/find_ef.php). skos:exactMatch, skos:related , and skos:closeMatch can then be used to link ECF classifications from different publishers.  We have previously discussed that the hierarchical categories (e.g., from BEIS) cannot be modelled as ontologies because subClassOf relationship doesn't always apply. And because we cannot change the classification systems of the publishers, SKOS could be the right solution to model these categories.

**Reply by dgarijo** [reply]
Tag is a skos:Concept, so we essentially support this.

**Reply by m-markovic** [reply]
Renaming to "Category" could still be beneficial to align with GHG domain terminology

**Reply by stefanogermano** [reply]
It might also be interesting to explore how these relate to 'activities' (with this term I mean the 'thing' an ECF refers to); even if we might want to keep the two concepts separate.
